test/gurka/test_shortcut.cc

```diff
@@ -74,6 +74,52 @@ TEST(Shortcuts, CreateInvalid) {
   EXPECT_ANY_THROW(gurka::findEdgeByNodes(graph_reader, layout, "C", "A"));
 }
 
+// Here no shortcuts are created.
+// The grid size should be greater than the max length that allows shortcuts to be created
+TEST(Shortcuts, CreateTooLong) {
+  constexpr double gridsize = 2000000;
+
+  const std::string ascii_map = R"(
+      A--B--C--D----E
+  )";
+
+  const gurka::ways ways = {
+      {"AB",
+       {{"highway", "primary"},
+        {"name", "Independence Avenue"},
+        {"maxspeed:forward", "80"},
+        {"maxspeed:backward", "80"}}},
+      {"BC",
+       {{"highway", "primary"},
+        {"name", "Independence Avenue"},
+        {"maxspeed:forward", "80"},
+        {"maxspeed:backward", "80"}}},
+       {"CD",
+       {{"highway", "primary"},
+        {"name", "Independence Avenue"},
+        {"maxspeed:forward", "60"},
+        {"maxspeed:backward", "60"}}},
+      {"DE",
+       {{"highway", "primary"},
+        {"name", "Independence Avenue"},
+        {"maxspeed:forward", "60"},
+        {"maxspeed:backward", "60"}}},
+  };
```

Comment:
still, I don't think this test is proving that the changes work now. you change properties going from `BC` to `CD`, so no matter the edge length it's not possible to have a shortcut from A -> D/E. to prove a point, all edges must have the same properties (you can only keep name & highway tag). 

and _ideally_ the map produces two shortcuts from that stretch (might be you have to extend it a little). the first one aborts due to length issues and starts a second one.